src/signing_commitment.rs

```diff
@@ -2,26 +2,75 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
+use crate::frost::keys::SigningShare;
+use crate::frost::round1::NonceCommitment;
+use crate::frost::round1::SigningCommitments;
+use crate::nonces::deterministic_signing_nonces;
+use crate::participant::Identity;
+use crate::participant::Secret;
+use crate::participant::Signature;
+use crate::participant::SignatureError;
+use crate::participant::IDENTITY_LEN;
+use siphasher::sip::SipHasher24;
+use std::borrow::Borrow;
+use std::error;
+use std::fmt;
+use std::hash::Hasher;
 use std::io;
 
-use reddsa::frost::redjubjub::round1::NonceCommitment;
+const NONCE_COMMITMENT_LEN: usize = 32;
+const CHECKSUM_LEN: usize = 8;
+pub const AUTHENTICATED_DATA_LEN: usize = IDENTITY_LEN + NONCE_COMMITMENT_LEN * 2 + CHECKSUM_LEN;
+pub const SIGNING_COMMITMENT_LEN: usize = AUTHENTICATED_DATA_LEN + Signature::BYTE_SIZE;
 
-use crate::participant::{Identity, Secret, Signature, SignatureError, IDENTITY_LEN};
+type Checksum = u64;
 
-const NONCE_COMMITMENT_LENGTH: usize = 32;
-pub const SIGNATURE_DATA_LENGTH: usize = IDENTITY_LEN + NONCE_COMMITMENT_LENGTH * 2;
-pub const SIGNING_COMMITMENT_LENGTH: usize = SIGNATURE_DATA_LENGTH + Signature::BYTE_SIZE;
+#[derive(Clone, Debug)]
+pub struct ChecksumError;
 
-fn signature_data(
+impl fmt::Display for ChecksumError {
+    #[inline]
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt::Display::fmt("checksum doesn't match", f)
+    }
+}
+
+impl error::Error for ChecksumError {}
+
+#[must_use]
+fn input_checksum<I>(transaction_hash: &[u8], signing_participants: &[I]) -> Checksum
+where
+    I: Borrow<Identity>,
+{
+    let mut signing_participants = signing_participants
+        .iter()
+        .map(Borrow::borrow)
+        .collect::<Vec<_>>();
+    signing_participants.sort_unstable();
+    signing_participants.dedup();
+
+    let mut hasher = SipHasher24::new();
```

Comment:
just curious - why this choice of hashing function?

Reply:
It's fast and modern. It's also the default hash function used by the Rust stdlib:

- https://doc.rust-lang.org/std/hash/struct.SipHasher.html
- https://doc.rust-lang.org/std/hash/struct.DefaultHasher.html

You might be wondering why am I using an external crate instead of using Rust's own implementation: the reason is that the default hash function used by Rust is not guaranteed to be stable across releases and may change in the future (see the documentation for DefaultHasher). This would be a problem because the same version of the Iron Fish SDK/client may be using a different hasher depending on what Rust compiler version was used.